src/lib/decoders.ts

```diff
@@ -0,0 +1,118 @@
+import { Result } from '@ethersproject/abi';
+import BigNumber from 'bignumber.js';
+import { BytesLike, defaultAbiCoder } from 'ethers/lib/utils';
+import { parseInt } from 'lodash';
+import { BN_0 } from '../bignumber-constants';
+import { NULL_ADDRESS } from '../constants';
+import { MultiResult } from './multi-wrapper';
+
+export function generalDecoder<T>(
+  result: MultiResult<BytesLike>,
+  types: string[],
+  defaultValue: T,
+  parser?: (v: Result) => T,
+): T {
+  if (!result.success || result.returnData === '0x') {
+    return defaultValue;
+  }
+  const decoded = defaultAbiCoder.decode(types, result.returnData);
+  return parser ? parser(decoded) : decoded[0];
+}
+
+export const uintDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return BigInt(defaultAbiCoder.decode(['uint'], result.returnData)[0]);
+};
+
+export const uint256ArrayDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return defaultAbiCoder
+    .decode(['uint256[]'], result.returnData)[0]
+    .map((r: BigNumber) => BigInt(r.toString()));
```

Comment:
The `BigNumber` type in ethers is not `bignumber.js` but `BN.js`, which is completely different, so you have the wrong type here, but since we're just converting to string you could use `any` here.

Reply:
Changed to `any`